### PR TITLE
Allow disabling of applicant auth

### PR DESCRIPTION
**Pull request by kmcnellis**
### Description

By setting CIVIFORM_APPLICANT_IDP to "disabled", remove all checks for auth provider configuration, and remove the main login button from the home screen.

### CODE REVIEW

File: server/app/views/LoginForm.java [hunk 1]
```diff
@@ -90,22 +94,23 @@ private ContainerTag mainContent(Messages messages) {
...
-                Styles.GAP_2,
-                Styles.PY_6,
-                Styles.PX_8,
-                Styles.TEXT_LG,
-                Styles.W_FULL,
-                Styles.PLACE_ITEMS_CENTER)
-            .with(p(loginMessage))
-            .with(loginButton(messages)));
-
+    if (applicantIdpName != AuthIdentityProviderName.DISABLED_APPLICANT) {
```
Comment (azizova-leyla) [inline server/app/views/LoginForm.java:97]: can we have a message saying "resident auth is disabled"?

File: server/app/views/LoginForm.java [hunk 2]
```diff
@@ -129,7 +134,7 @@ private ContainerTag mainContent(Messages messages) {
     }
     content.with(alternativeLoginButtons);
 
-    String somethingElse = messages.at(MessageKey.CONTENT_ADMIN_LOGIN_PROMPT.getKeyName());
```
Comment (azizova-leyla) [inline server/app/views/LoginForm.java:132]: that was... something else =)
thanks for renaming

**Review by bion — CHANGES_REQUESTED** [review]

File: server/app/views/LoginForm.java [hunk 3]
```diff
@@ -38,7 +40,9 @@ public class LoginForm extends BaseHtmlView {
   public LoginForm(BaseHtmlLayout layout, Config config, FakeAdminClient fakeAdminClient) {
     this.layout = checkNotNull(layout);
     this.applicantIdp = checkNotNull(config).getString("auth.applicant_idp");
-    this.idcsIsAvailable = checkNotNull(config).hasPath("idcs.register_uri");
+    this.applicantIdpName = AuthIdentityProviderName.forString(applicantIdp).get();
+    this.idcsIsAvailable = (this.applicantIdpName == AuthIdentityProviderName.IDCS_APPLICANT
+        && checkNotNull(config).hasPath("idcs.register_uri"));
```
[the remark this answers is not in the document]
Comment (kmcnellis) [reply server/app/views/LoginForm.java:45]: This is still IDCS-specific - it is for the IDCS register URL.  I'm just also disabling it when auth is disabled, even if the value is present in the config

[the remark this answers is not in the document]
Comment (kmcnellis) [reply server/app/views/LoginForm.java:45]: Also, this is not a string.  It's an enum

File: server/app/views/LoginForm.java (hunk 1 above)
Comment (kmcnellis) [reply server/app/views/LoginForm.java:97]: Done

File: server/app/views/LoginForm.java (hunk 3 above)
[the remark this answers is not in the document]
Comment (kmcnellis) [reply server/app/views/LoginForm.java:45]: Updated to make it all a bit more clear

File: server/app/views/LoginForm.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (kmcnellis) [reply server/app/views/LoginForm.java:97]: done

File: server/app/views/LoginForm.java (hunk 2 above)
Comment (kmcnellis) [reply server/app/views/LoginForm.java:132]: I think it's because the message is "Looking for something else?", haha

**Comment by kmcnellis** [comment]
image

**Review by bion — APPROVED** [review]

**Review by azizova-leyla — APPROVED** [review]

File: server/app/views/LoginForm.java [hunk 4]
```diff
@@ -37,8 +39,8 @@ public class LoginForm extends BaseHtmlView {
   @Inject
   public LoginForm(BaseHtmlLayout layout, Config config, FakeAdminClient fakeAdminClient) {
     this.layout = checkNotNull(layout);
+    checkNotNull(config);
```
[the remark this answers is not in the document]
Comment (kmcnellis) [reply server/app/views/LoginForm.java:42]: done

**Pull request merged**